### DISCUSSION TITLE
Breaking changes changelog

### DISCUSSION THREAD

**Discussion by k0shk0sh** [supabase/supabase#2781] [General]
I would definitely recommend when updating your libraries which introduces breaking changes to provide a migration process. 

for example

supabase-flutter in the library it had a function which was returning `Future<bool> hasAccessToken()` and after I updated the library prior to `0.1.0`, basically there was no where I could find it and nowhere in the changelog it was mentioned. 

just a tip for the future tho :slight_smile: especially with supabase-flutter/supabase-dart 0.1.0 I see it has few breaking changes but nowhere in the changelog it mentioned the breaking changes. 

> [0.1.0]
   chore: update supabase to v0.1.0
   feat: using hive to persist session by default
 

this is definitely a bad developer experience and I hope this changes in the future. 

thanks

P.S: this is also posted in https://discord.com/channels/839993398554656828/839993398554656831/874309583357763685

**Comment by dshukertjr** [comment]
Hi @k0shk0sh!

Thanks for the feedback. So sorry about not labeling 0.1.0 as breaking change. That is on me, and we will take better care in the future to prevent something like this from happening again. Again, these feedbacks are great, so if you have any in the future, please don't hesitate to let us know!
